engine/interpreter/src/main/scala/pl/touk/nussknacker/engine/compile/PartSubGraphCompiler.scala

```diff
@@ -239,22 +248,26 @@ class PartSubGraphCompiler(expressionCompiler: ExpressionCompiler,
 
 object PartSubGraphCompiler {
 
-  private case class ExpressionTypingResult(typingResult: TypingResult, typingInfo: Option[ExpressionTypingInfo]) {
+  object ExpressionTypingResult {
+    val unknown: Map[String, ExpressionTypingInfo] =
+      ExpressionTypingResult(Unknown, None).toDefaultExpressionTypingInfoEntry.toMap
+
+    def apply(typedExpression: TypedExpression): ExpressionTypingResult =
+      ExpressionTypingResult(typedExpression.returnType, Some(typedExpression.typingInfo))
+  }
+  case class ExpressionTypingResult(typingResult: TypingResult, typingInfo: Option[ExpressionTypingInfo]) {
 
     def toDefaultExpressionTypingInfoEntry: Option[(String, ExpressionTypingInfo)] =
       typingInfo.map(NodeTypingInfo.DefaultExpressionId -> _)
 
   }
 
-  private case class FieldExpressionTypingResult(fieldName: String, private val exprTypingResult: ExpressionTypingResult) {
+  case class FieldExpressionTypingResult(fieldName: String, private val exprTypingResult: ExpressionTypingResult) {
```

Comment:
I think we should extract it outside PartSubGraphCompiler if we want to use it outside